main/src/ca/uwaterloo/flix/language/ast/TokenKind.scala

```diff
@@ -326,6 +326,16 @@ sealed trait TokenKind {
     case _ => false
   })
 
+  /**
+    * Checks if this token is one of the [[TokenKind]]s that can validly appear after a doc-comment.
+    * Doc-comments may only annotate declarations and enum-cases.
```

Comment:
and laws, apparently :)

Reply:
Well a `law` is a declaration! In the eyes of the parser at least.